src/effects/sinkSuspense.ts

```diff
@@ -3,45 +3,79 @@ import { SUSPENSE } from "../SUSPENSE"
 import { sinkSuspense as ISinkSuspense } from "../index.d"
 
 type SubscriberWithInner<T> = Subscriber<T> & { inner: Subscriber<any> }
+
+/**
+ * `sinkSuspense()` and `liftSuspense() are operators that help deal with
+ * SUSPENSE values on the streams, which is useful when the meaning of SUSPENSE
+ * is that everything needs to be reset.
+ *
+ * `sinkSuspense()` is an operator that when it receives a SUSPENSE, it will
+ * throw it as an error down the stream, which resets all of the observables
+ * down below. It will then hold the subscription to the upstream, waiting for a
+ * resubscription to happen immediately. If it doesn't happen, then it will
+ * unsubscribe from upstream.
+ *
+ * ```ts
+ * const account$ = accountSwitch$.pipe(switchMapSuspended((v) => fetchAccount(v)))
+ *
+ * const posts$ = account$.pipe(
+ *   switchMap((v) => (v === SUSPENSE ? of(SUSPENSE) : fetchPosts(v))),
+ * )
+ *
+ * // with sinkSuspense
+ * const account$ = accountSwitch$.pipe(
+ *   switchMapSuspended((v) => fetchAccount(v)),
+ *   sinkSuspense(),
+ * )
+ *
+ * const posts$ = account$.pipe(switchMap((v) => fetchPosts(v)))
+ * ```
+ *
+ * It's very important to remember that `sinkSuspense` is throwing SUSPENSE
+ * values as errors, which means that subscriptions will get closed, in ways
+ * that are not always obvious. In most of the cases, it can be solved by using
+ * `liftSuspense()`, dealing with that value, and calling `sinkSuspense()`
+ * again. Use at your own risk.
```

Comment:
Thanks a lot @bruceharris ! However, it's actually a bit more nuanced than this.

Unfortunately, I'm currently under a very heavy workload and I won't be able to get to this until 2-3 weeks from now. All this is to say that it's not like I'm ignoring this, it's that it will take me a few weeks until I'm able to make time for it.

Reply:
Actually, now that I think about it, the release notes explain it fairly well:

> `sinkSuspense()` is an operator that when it receives a SUSPENSE, it will throw it as an error down the stream, which resets all of the observables down below. It will then hold the subscription to the upstream, waiting for a resubscription to happen immediately. If it doesn't happen, then it will unsubscribe from upstream.

Sorry, but I really dislike the description that you put in your comment.

Reply:
Anyways, what I mean is that I would much rather explain how to use it.

I'm aware that we put those comments in the release notes. However, we did that because we currently lack a proper set of docs for this low-level library and we didn't want for people to use this operator without the proper understanding. However, the comments on the code-base should help people who read the codebase understand what the code is doing, that's why I'm not ok adding these comments into the codebase. 

I hope that makes sense.